actionmailbox/app/controllers/action_mailbox/ingresses/amazon/inbound_emails_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module ActionMailbox
+  # Ingests inbound emails from Amazon SES/SNS and confirms subscriptions.
+  #
+  # Subscription requests must provide the following parameters in a JSON body:
+  # - +Message+: Notification content
+  # - +MessageId+: Notification unique identifier
+  # - +Timestamp+: iso8601 timestamp
+  # - +TopicArn+: Topic identifier
+  # - +Type+: Type of event ("SubscriptionConfirmation")
+  #
+  # Inbound email events must provide the following parameters in a JSON body:
+  # - +Message+: Notification content
+  # - +MessageId+: Notification unique identifier
+  # - +Timestamp+: iso8601 timestamp
+  # - +TopicArn+: Topic identifier
+  # - +Type+: Type of event ("Notification")
+  #
+  # All requests are authenticated by validating the provided AWS signature.
+  #
+  # Returns:
+  #
+  # - <tt>204 No Content</tt> if a request is successfully processed
+  # - <tt>401 Unauthorized</tt> if a request does not contain a valid signature
+  # - <tt>404 Not Found</tt> if the Amazon ingress has not been configured
+  # - <tt>422 Unprocessable Entity</tt> if a request provides invalid parameters
+  #
+  # == Usage
+  #
+  # 1. {Configure SES}[https://docs.aws.amazon.com/ses/latest/DeveloperGuide/receiving-email-notifications.html] to route emails through SNS. Take note of the topic unique reference (+TopicArn+).
+  #    The option "Enable raw message delivery" should not be selected. See {documentation}[https://docs.aws.amazon.com/sns/latest/dg/sns-large-payload-raw-message-delivery.html] for more details.
+  #
+  #    {Configure SNS}[https://docs.aws.amazon.com/ses/latest/DeveloperGuide/receiving-email-action-sns.html] to send notifications to +/rails/action_mailbox/amazon/inbound_emails+.
+  #
+  #    If your application is found at <tt>https://example.com</tt> you would specify the fully-qualified URL <tt>https://example.com/rails/action_mailbox/amazon/inbound_emails</tt>.
+  #
+  # 2. Install the {aws-sdk-sns}[https://rubygems.org/gems/aws-sdk-sns] gem:
+  #
+  #        # Gemfile
+  #        gem "aws-sdk-sns", "~> 1.9", require: false
+  #
+  # 3. Tell Action Mailbox to accept notifications from Amazon:
+  #
+  #        # config/environments/production.rb
+  #        config.action_mailbox.ingress = :amazon
```

Comment:
I think that this is not future proof from a product standpoint. I understand that Rails ActiveStorage today uses the :amazon key name as well, which makes sense why it's carried forward, but now :amazon is canned into using one implementation.

In this case, what if instead of SNS, a Lambda implementation comes along? SES would forward the email to Lambda which calls the webhook. Sounds great and could possibly get around the message size limit. But now what do we call this option if not :amazon?

One of the challenges of the AWS SDK is the growing list of services, and sometimes those services can share overlap in functionality. Because of this, in aws-sdk-rails, I ended up changing the SES mailer's configuration to :ses rather than :amazon. This is knowing that we might want different options for SESv2 or Pinpoint.

I'm not sure where the Rails team stands on this when it comes to AWS functionality, but personally I'd like to see this and things like ActiveStorage have configuration of the product name (config as :s3) and not the company name, but I understand not making this decision to be consistent.

Reply:
@mullermp When you say lambda in this context do you mean SES with a lambda action or straight lambda?

fwiw I think that aws or amazon should be part of the naming because those were the names I went looking for when I was trying to find this functionality.  So if we go with more specialized naming I'd suggest `:amazon_ses`

Reply:
Sorry, I meant SES with a lambda action. For search-ability, that's a fair point. :)